### DISCUSSION TITLE
fastAPI uvicorn works on localhost but fails on deployement.

### DISCUSSION THREAD

**Discussion by yecohn** [fastapi/fastapi#9610] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
...
```

### Description

Hi, 

I am trying to deploy a simple app with FastAPI. when running my app on localhost there is no issue. 

However when I deploy on a server: 
`uvicorn app:app --host 0.0.0.0 --port 8001`

I cannot connect to the app: 
 `curl http://<ip_adress_server>:8001`
I get an error: 

"""
This page isn’t working

<ip_adress>  didn’t send any data.
ERR_EMPTY_RESPONSE

""" 

I do not want to use Nginx. I am wondering what is going on ? 

thanks 

### Operating System

Linux

### FastAPI Version

0.95

### Python Version

3.8

**Comment by n8sty** [comment, marked as answer]
It's impossible to help you debug without a minimally reproducible example but my guess is that you aren't correctly mapping the port FastAPI is running on so that it can handle requests.

**Comment by chrisK824** [comment]
Indeed not much information here, but from available info it seems you need to open that port in server's firewall so you can access it from outer world. Or if you are using a DNS then this is not configured properly

**Comment by vanishingESCkey** [comment]
For development in private network I have used such command

`uvicorn app.main:app --reload --host myhostname --port 3210`

Then I was able to see it in browser and share to my coworkers using uri http://myhostname:3210

**Comment by abhijit673** [comment]
I tried with the following command,

`uvicorn app:app --host 0.0.0.0 --port 8000`

Initially I could not access the API with the public IP but after opening the port from the cloud network and also from the VM, it worked just fine.
